cmd/application.go

```diff
@@ -47,14 +47,31 @@ var applicationCreateCmd = &cobra.Command{
 	Example: `  # Create an application
   odo app create myapp
 	`,
-	Args: cobra.ExactArgs(1),
+	Args: cobra.RangeArgs(0, 1),
 	Run: func(cmd *cobra.Command, args []string) {
-		// Args validation makes sure that there is exactly one argument
-		name := args[0]
+		var name string
+		client := getOcClient()
+		if len(args) == 1 {
+			// The only arg passed is the app name
+			name = args[0]
+		} else {
+			// Desired app name is not passed so, generate a new app name
+			// Fetch existing list of apps
+			apps, err := application.List(client)
+			checkError(err, "")
+			appNames := []string{}
+			// Collect the names of existing apps to avoid generated name being already used
+			for _, app := range apps {
```

Comment:
Seems a bit weird that we are using a loop to append appNames and then passing it into `GetDefaultAppName`.

Wouldn't it be better to pass in `apps`, the previous variable you used, into a function that checks to see that the current app doesn't exist? Passing in the struct or whatever parameter `apps` is into another function makes more sense than using this loop.

Reply:
Ok so you are saying, instead of getting names of apps from []config.ApplicationInfo here, do it inside GetDefaultAppName...  right? Ok

Reply:
I mean pass in the `apps` variable directly to GetDefaultAppName and iterate through it there.

Reply:
Ack done.. cahnged it to the same(now)..

Note: https://github.com/redhat-developer/odo/pull/741/files#diff-6c6a7fefe37261e2fa8a010cf9e7ae08R70 is very generic in nature and used for both component name generation and app name generation. So, I am formulating set of names in respective function(GetDefaultAppName in this case) before delegating to https://github.com/redhat-developer/odo/pull/741/files#diff-6c6a7fefe37261e2fa8a010cf9e7ae08R70